clientd/src/lib.rs

```diff
@@ -12,18 +14,55 @@ use tokio::sync::{broadcast, mpsc, oneshot};
 mod rpc;
 pub mod server;
 
-// TODO: remove pub and use this type to enforce invariants
-#[derive(Debug, Serialize)]
-pub struct CallbackResponse(pub Result<String, RpcError>);
+#[derive(Clone)]
+pub struct EventSubscribers(HashMap<EventKey, broadcast::Sender<Event>>);
 
-// TODO: figure out best approach for error handling
-#[derive(Debug, Serialize)]
-pub enum RpcError {
-    ClientError,
+impl EventSubscribers {
+    pub fn new() -> Self {
+        let mut se_map = HashMap::new();
```

Comment:
`se_`? :)

Reply:
Probably for `self`. Just `map` would work, it's not a keyword or anything.

Reply:
"subscriber_event_map"..yeah I was not sure about that name but it sounds professional haha. should I change it to the longer version ?

Reply:
`se` is just not very typical. I see people either doing `s`, or `map` or `self_` or `r#self`